### PR TITLE
Added JSDoc style documentation to index.ts

### LINKED ISSUE #38: Add JSDoc comments to index.ts
Currently, the functions in `index.ts` lack JSDoc comments. We should add them.

This will help us move towards good API docs for `commutable`.

### PR DESCRIPTION
**Pull request by jdetle**
Fixes #38.
Hopefully this is somewhat up to muster. I am unfamiliar with TypeScript so this has been a fun night of learning a new language. I consistently reference 'Map representation', if there is a more appropriate name for this object please let me know. I appreciate the opportunity to get to make a contribution!

### CODE REVIEW

**Comment by captainsafia** [comment]
Thanks for opening up this PR, @jdetle, and welcome to the nteract community!

I've posted some comments inline about the code you've submitted.

File: src/index.ts [hunk 1]
```diff
@@ -2,13 +2,27 @@ import {
   fromJS as _fromJS,
   List,
   Map,
-} from 'immutable';
+} from 'Map';
```
Comment (captainsafia) [inline src/index.ts]: The `Map` object that is referenced throughout the file is actually the `Map` object that is imported in line 4. We use a library called [Immutable](https://facebook.github.io/immutable-js/) to store our data structures. 

Immutable contains the objects `List` and `Map` so we use the `import { ... } from ...` syntax to import them them into the context of our program. Whenever we refer to `Map` or `List` we are referring to the objects provided by immutable.

In summary, this should be `from 'immutable'` as it was before.

Comment (jdetle) [inline src/index.ts]: This was a search and replace error! Haha fixing it right now.
On Fri, Jun 10, 2016 at 12:50 AM Safia Abdalla <notifications@github.com>
wrote:

> In src/index.ts
> <https://github.com/nteract/commutable/pull/39#discussion_r66573966>:
>
> > @@ -2,13 +2,27 @@ import {
> >    fromJS as _fromJS,
> >    List,
> >    Map,
> > -} from 'immutable';
> > +} from 'Map';
>
> The Map object that is referenced throughout the file is actually the Map
> object that is imported in line 4. We use a library called Immutable
> <https://facebook.github.io/immutable-js/> to store our data structures.
>
> Immutable contains the objects List and Map so we use the import { ... }
> from ... syntax to import them them into the context of our program.
> Whenever we refer to Map or List we are referring to the objects provided
> by immutable.
>
> In summary, this should be from 'immutable' as it was before.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nteract/commutable/pull/39/files/37ab2487875cfb78128027740d7b84c04abf1978#r66573966>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKcARtraVuP22QGsiQHqNMCNMxRofGOYks5qKRc4gaJpZM4IysXf>
> .
>
-- 
Have a wonderful day,
John J. Detlefs

Comment (jdetle) [inline src/index.ts]: Thanks for the explanation though, I had no idea where the immutable package was from!

**Comment by captainsafia** [comment]
Your PR has been merged! Welcome to the nteract community!

By the way, come hang out on [our Slack](http://slack.nteract.in/) if you are interested in contributing further or seeing what we are up to!

**Comment by jdetle** [comment]
Ahhhh so exciting! I definitely will. Thanks again.
On Fri, Jun 10, 2016 at 12:57 AM Safia Abdalla <notifications@github.com>
wrote:

> Merged #39 <https://github.com/nteract/commutable/pull/39>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nteract/commutable/pull/39#event-688244198>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AKcARpY8ErVLwHKlu0UimzmY_Y__7Yvnks5qKRjbgaJpZM4IysXf>
> .
>
-- 
Have a wonderful day,
John J. Detlefs

**Comment by rgbkrk** [comment]
Thanks for the contribution @jdetle! Added you to the org. Thanks for joining up!
